rust/shalom_dart_codegen/src/lib.rs

```diff
@@ -93,18 +94,33 @@ mod ext_jinja_fns {
         }
     }
 
-    pub fn parse_field_default_value(input: ViaDeserialize<InputFieldDefinition>) -> String {
-        let default_value = input.0.default_value;
+    pub fn parse_field_default_value(
+        schema_ctx: &SchemaContext,
+        input: ViaDeserialize<InputFieldDefinition>,
+    ) -> String {
+        let input = input.0;
+        let default_value = &input.default_value.as_ref();
+
         if default_value.is_none() {
             panic!("cannot parse default value that does not exist")
         }
         let default_value = default_value.unwrap();
-        default_value.to_string()
+        let ty = input.resolve_type(schema_ctx);
+        if let GraphQLAny::Enum(enum_) = ty {
+            format!("{}.{}", enum_.name, default_value)
+        } else {
+            default_value.to_string()
+        }
     }
 
-    pub fn is_input_type(schema_ctx: &SchemaContext, ty: ViaDeserialize<FieldType>) -> bool {
+    pub fn input_type(schema_ctx: &SchemaContext, ty: ViaDeserialize<FieldType>) -> String {
         let ty = schema_ctx.get_type(&ty.0.name()).unwrap();
-        matches!(ty, GraphQLAny::InputObject(_))
+        match ty {
+            GraphQLAny::Scalar(_) => "Scalar".to_string(),
+            GraphQLAny::InputObject(_) => "InputObject".to_string(),
+            GraphQLAny::Enum(_) => "Enum".to_string(),
+            _ => unimplemented!("input type not supported"),
+        }
     }
```

Comment:
this is useless and should be avaialbe by default because we do 
`#[serde(tag = "kind")]` on it